Kaltura/com.tle.core.kaltura/src/com/tle/core/kaltura/service/KalturaServiceImpl.java

```diff
@@ -429,16 +433,23 @@ public void enable(KalturaServer ks, boolean enable) {
     kalturaDao.update(ks);
   }
 
-  @Override
-  public boolean hasConf(KalturaServer ks, String confId) {
-    return getUiConfig(ks, confId).isPresent();
-  }
-
   @Override
   public UiConf getPlayerConfig(KalturaServer ks, String confId) {
-    return getUiConfig(ks, confId).orElseThrow(
-            () -> new RuntimeException("Failed to get Kaltura player details for " + confId)
-    );
+    try {
+      int playerId = Optional.ofNullable(confId)
+          .filter(StringUtils::isNotEmpty)
+          .map(Integer::parseInt)
+          .orElseGet( () -> {
+            LOGGER.debug("No conf ID provided, use the ID configured in Kaltura setting instead.");
+            return ks.getKdpUiConfId();
+          });
+
+      Client client = getKalturaClient(ks, SessionType.ADMIN);
+      return execute(UiConfService.get(playerId).build(client));
+    } catch (APIException | NumberFormatException e) {
+      LOGGER.error("Failed to get Kaltura player details for " + confId + ", use the OEQ default player instead.", e);
```

Comment:
Should this possibly just be a warning? Also, there's a slight typo:

```suggestion
      LOGGER.warn("Failed to get Kaltura player details for " + confId + ", using the oEQ default player instead.", e);
```

Reply:
Thanks Ian. Updated.